packages/theme/components/ValidationProviders/EmailValidationProvider.vue

```diff
@@ -0,0 +1,49 @@
+<template>
+  <ValidationProvider
+    v-slot="{ errors }"
+    :rules="mergedRules"
+  >
+    <slot
+      :errors="errors"
+    />
+  </ValidationProvider>
+</template>
+
+<script>
+import { ValidationProvider } from 'vee-validate';
+import { computed } from '@nuxtjs/composition-api';
+
+export default {
+  name: 'EmailValidationProvider',
+  props: {
+    name: {
+      type: String,
+      default() {
+        return 'email';
+      }
+    },
+    rules: {
+      type: String,
+      default() {
+        return 'email';
+      }
+    },
+    'append-rules': {
+      type: String,
+      default() {
+        return '';
+      }
+    }
+  },
+  components: {
+    ValidationProvider
+  },
+  setup({ rules, appendRules }) {
+    const mergedRules = computed(() => `${rules}${appendRules ? `|${appendRules}` : ''}`);
```

Comment:
Let's make this a reusable utility to avoid having the same logic duplicated.

Maybe a composable that takes a [rest](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters) parameter "rules" and returns computed "rules" string?
Usage: `const { mergedRules } = useMergedValidationRules(props.rules, props.appendRules)`?

I believe, useMergedValidationRules could join the rest argument with "|", then split('|').filter(Boolean).join('|') to make sure all rules are given in proper format. This is just an idea/hint to focus on input sanitization. Implementation is up to you.

Reply:
The template string solution was simpler with only two values to merge (rules, appendRules). If we go with the composable that takes a variable amount of arguments then using join seems more reasonable. Let's do it!